packages/server/src/fhir/search.ts

```diff
@@ -143,6 +148,29 @@ async function getSearchEntries<T extends Resource>(
   };
 }
 
+function removeResourceFields(resource: Resource, repo: Repository, searchRequest: SearchRequest): void {
+  repo.removeHiddenFields(resource);
+  if (searchRequest.fields) {
```

Comment:
I believe that `fields` and `elements` need to be tracked separately in `SearchRequest`.

Currently, `fields` is a mix of element names and search parameter codes.  For example, when looking at `Patient`, you can add `id`, `name`, and `telecom` (all element names) but you can also add `email` and `phone` which are not element names but instead search parameters.

On the client, we evaluate the search parameter to show the value.

https://app.staging.medplum.com/Patient?_count=20&_fields=id,_lastUpdated,name,birthDate,gender,telecom,email,phone&_offset=0&_sort=-_lastUpdated

<img width="948" alt="image" src="https://github.com/medplum/medplum/assets/749094/960eaa68-85fa-41ba-aedd-3580639bf29d">